src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1277 @@
+
+        # ---------------------------------------------------------------------
+
+        # # # PROGRESS TO SEVERE WASTING # # # # # # # # # # # # # # # # # #
+        # Determine those that will progress to severe wasting (WHZ < -3) and schedule progression event ---------
+        mod_wasting = df.loc[df.is_alive & (df.age_exact_years < 5) &
+                             (df.un_WHZ_category == '-3<=WHZ<-2')]
+        progression_severe_wasting = self.module.wasting_models.severe_wasting_progression_lm.predict(
+            mod_wasting, rng=rng, squeeze_single_row_output=False)
+
+        # determine those individuals who will progress to severe wasting and time of progression
+        for person in mod_wasting.index[progression_severe_wasting]:
+            outcome_date = self.module.date_of_outcome_for_untreated_wasting(person_id=person)
+            # schedule severe wasting WHZ < -3 onset
+            if outcome_date <= self.sim.date:
+                # schedule severe wasting (WHZ < -3) onset today
+                self.sim.schedule_event(event=ProgressionSevereWastingEvent(
+                    module=self.module, person_id=person), date=self.sim.date)
+            else:
+                # schedule severe wasting WHZ < -3 onset according to duration
+                self.sim.schedule_event(
+                    event=ProgressionSevereWastingEvent(
+                        module=self.module, person_id=person), date=outcome_date)
+
+        # # # MODERATE WASTING NATURAL RECOVERY # # # # # # # # # # # # # #
+        # Schedule recovery from moderate wasting for those not progressing to severe wasting ---------
+        for person in mod_wasting.index[~progression_severe_wasting]:
+            outcome_date = self.module.date_of_outcome_for_untreated_wasting(person_id=person)
+            if outcome_date <= self.sim.date:
+                # schedule recovery for today
+                self.sim.schedule_event(event=WastingNaturalRecoveryEvent(
+                    module=self.module, person_id=person), date=self.sim.date)
+            else:
+                # schedule recovery according to duration
+                self.sim.schedule_event(event=WastingNaturalRecoveryEvent(
+                    module=self.module, person_id=person), date=outcome_date)
+
+        # ------------------------------------------------------------------------------------------
+        # ## UPDATE PROPERTIES RELATED TO CLINICAL ACUTE MALNUTRITION # # # #
+        # ------------------------------------------------------------------------------------------
+        # This applies to all children under 5
+
+        # give MUAC measurement category for all WHZ, including well nourished children -----
+        # determine the presence of bilateral oedema / oedematous malnutrition
+        # determine the clinical state of acute malnutrition, and check complications if SAM
+        self.module.population_poll_clinical_am(df)
```

Comment:
@emmanuel, @thewati, So every month, the muac measurement category, and presence of oedematous are re-determined for all children?

Reply:
Are you suggesting that we should determine the muac category and oedematous presence at incidence, progress to severe wasting, natural recovery and recovery to MAM(these are the times we expect WHZ to change)?

Reply:
Yes, exactly. It might be an issue if you scheduled someone for treatment only based on MUAC or oedema. If their WHZ didn’t change but MUAC/oedema did, they may suddenly not be diagnosed with MAM/SAM. However, they are already receiving treatment, but the probabilities will make someone else be diagnosed with MAM/SAM, which may lead to them being sent for treatment, thus increasing the pool of children needing treatment. This is happening every month. I didn’t check the numbers on how often something like that would happen, but it has the potential to change the number of treated children. If we only update MUAC/oedema when WHZ changes, it will preserve the MAM/SAM diagnoses for those on treatment until recovery or death.

Reply:
@emmanuel, now the MUAC, presence of oedema,  clinical state of acute malnutrition, and complications if SAM are updated only when population is initiated or when WHZ is changed and the other anthropometric indices are not set, see the commit [21154414].